### PR TITLE
Allow specifying alternate MediaBaseURL at runtime

### PR DESCRIPTION
**Pull request by t3chguy**
`--media-base-url="https://matrix.org"`

### CODE REVIEW

File: src/github.com/t3chguy/matrix-static/matrix-static.go [hunk 1]
```diff
@@ -266,7 +274,7 @@ func main() {
 				AtBottomEnd: jobResult.AtBottomEnd,
 
 				Sanitizer:         sanitizerFn,
-				HomeserverBaseURL: client.HomeserverURL.String(),
+				HomeserverBaseURL: config.MediaBaseURL,
```
Comment (richvdh) [inline src/github.com/t3chguy/matrix-static/matrix-static.go]: surely "HomeserverBaseURL" should still be the HomeserverURL ?

File: src/github.com/t3chguy/matrix-static/mxclient/mxclient.go [hunk 2]
```diff
@@ -40,7 +40,10 @@ type RespInitialSync struct {
 }
 
 // Our Client extension adds some methods
```
Comment (richvdh) [inline src/github.com/t3chguy/matrix-static/mxclient/mxclient.go]: ... and some fields, apparently

File: src/github.com/t3chguy/matrix-static/matrix-static.go (hunk 1 above)
Comment (t3chguy) [reply src/github.com/t3chguy/matrix-static/matrix-static.go]: Well this is the URL being passed to the HTML template, as far as the end-users are concerned they should only ever see the MediaBaseURL because the HomeserverURL may not resolve for them, as is in the case of view.matrix.org

**Comment by richvdh** [comment]
Perhaps the field needs renaming then?

**Comment by t3chguy** [comment]
> so why does this have to be set on the commandline rather than being read from the config file?

The config is currently entirely about how to connect to Matrix, all the hosting-specific stuff is provided at runtime via arguments.

**Comment by t3chguy** [comment]
Changed the field name as suggested

File: src/github.com/t3chguy/matrix-static/matrix-static.go [hunk 3]
```diff
@@ -84,12 +88,16 @@ func main() {
...
-	client, err := mxclient.NewClient(config.ConfigFile)
+	client, err := mxclient.NewClient(config.MediaBaseURL, config.ConfigFile)
 	if err != nil {
 		log.WithError(err).Error("Unable to start new Client")
 		return
 	}
 
+	if config.MediaBaseURL == "" {
+		config.MediaBaseURL = client.HomeserverURL.String()
+	}
```
Comment (krombel) [inline src/github.com/t3chguy/matrix-static/matrix-static.go]: Why not moving this a few lines up (somewhere before mxclient.NewClient) and remove [this](https://github.com/t3chguy/matrix-static/pull/59/files#diff-a342b5501fd6aed8e49147d8030ad39dR130) for deduplication?

Comment (t3chguy) [reply src/github.com/t3chguy/matrix-static/matrix-static.go]: because `client.HomeserverURL` is not defined until after `mxclient.NewClient`, and `NewClient` is what reads the config file which holds that value, unless im missing something?

**Comment by richvdh** [comment]
Firstly, sorry for sitting on this for so long.

> The config is currently entirely about how to connect to Matrix, all the hosting-specific stuff is provided at runtime via arguments.

I'm afraid I don't really buy this. I appreciate that we aren't really connecting to the MediaBaseURL, but it feels as much a thing that is about connecting to the matrix server as is HomeserverURL. I also think that this change would be much simpler if the setting was in the config file.

**Comment by t3chguy** [comment]
`+43 -27` vs `+63 -31` but sure :P

**Comment by t3chguy** [comment]
@richvdh back to you, can't assign you

File: src/github.com/t3chguy/matrix-static/templates/room-chat.qtpl [hunk 4]
```diff
@@ -358,7 +358,7 @@
                     numEvents := len(p.Events)
                 %}
                 {% for i, event := range p.Events %}
-                    {%= p.printEvent(&event, &prevEv, i == numEvents - 1) %}
+                    {%= p.printEvent(&event, &prevEv, p.Highlight && i == numEvents - 1) %}
```
Comment (richvdh) [inline src/github.com/t3chguy/matrix-static/templates/room-chat.qtpl]: ? looks unrelated ?

**Comment by richvdh** [comment]
> +43 -27 vs +63 -31 but sure :P

bah, fair. I hadn't realised you were abusing RespRegister as the config file format :-p

**Comment by t3chguy** [comment]
Removed unrelated thing
